data/org.freedesktop.portal.Notification.xml

```diff
@@ -106,21 +180,361 @@
+
+          * ``os.battery.low``
+
+            Intended to be used to indicate that the system is low on battery.
+
+          * ``browser.web-notification``
+
+            Intended to be used by browsers to mark notifications as web notifications
+
+          Additional types may be defined by notification servers.
+
+        * ``sound`` (``v``)
+
+        A serialized sound to add to the notification.
+        Supported image formats are oga and wav.
+
+        The format for
+        serialized sound is a tuple (sv) with the following supported keys:
+
+        * ``bytes`` (``ay``)
+
+          A bytes sound containing a bytestring with the sound data.
+
+          This is the same format as a serialized `GBytesIcon
+          <https://docs.gtk.org/gio/class.BytesIcon.html>`_.
+
+        * ``file-descriptor`` (``h``)
+
+          A file descriptor to a sound file.
+
+        * ``silent``
+
+          A "silent" sound indicates that the system setting should be ignored and
+          no sound should be played for the notification.
+
+        There may be further restrictions on the supported kinds of sounds.
+
+        * ``vibration`` (``au``)
+
+          Vibration pattern is an array of uint32 to describe
+          alternating periods in which the device should vibrate and not vibrate in milliseconds.
+          This array can contain an odd number of items since the last non-vibrating period can be omitted.
+
+          An empty array indicates to ignore system settings and don't vibrate.
+
+          This is a similar API to `Web Vibration API <https://developer.mozilla.org/en-US/docs/Web/API/Vibration_API>`_.
+
+        * ``led-color`` (``(yyyy)``)
+
+          Notification LED color as 4 bytes value ``(red, green, blue, alpha)``.
+
+          A tuple of ``(0000)`` indicates to ignore system settings and don't turn on the LED.
+
+        * ``display-hint`` (``as``)
+
+          An array of ways to display the notification. If none are set, or the notification server has its own policy, it is
+          free to decide how and where to display the notification.
+
+          * ``transient``
+
+            The notification is displayed only as a banner and won't be kept
+            by the server in a tray.
+
+            It's a programmer error to specify ``tray`` at the same time.
+
+          * ``tray``
+
+            No banner for the notification will be displayed and
+            the notification is placed in the tray.
+
+            It's a programmer error to specify ``transient`` at the same time.
+
+          * ``persistent``
+
+            Make the notification persistent in the notification tray.
+            The user can’t dismiss it using the usual close button or gesture.
+
+            Apps are only allowed to display persistent notifications
+            as long as they have a window. Once the last window of an app
+            is closed the persistent notification will be removed.
```

Comment:
- "tray" and "persistent": Apps shouldn't have control over notification UX. If you don't agree, please give example cases for those.
- "transient": Also part of notification UX. However, it seems to be a content type to display, e.g., an action like paste has been done (here equivalent of in-app notification). I don't know what really to think about this, so examples are also welcome here.